test/common/stats/thread_local_store_test.cc

```diff
@@ -381,6 +381,49 @@ TEST_F(StatsThreadLocalStoreTest, BasicScope) {
   store_->shutdownThreading();
   scope1->deliverHistogramToSinks(h1, 100);
   scope1->deliverHistogramToSinks(h2, 200);
+  scope1.reset();
+  tls_.shutdownThread();
+}
+
+TEST_F(StatsThreadLocalStoreTest, HistogramScopeOverlap) {
```

Comment:
Do we have any tests that cover the actual threading? Would it be possibly to build some with the synchronizer?